### ISSUE TITLE
[V4] GraphQL Union Types, Dynamic Zone `Cannot read property 'attributes' of undefined`

### ISSUE THREAD

**Issue by ScottAgirs** [strapi/strapi#11605]
### Describe the bug

I've created 
- `Single Type` called `Home Page`
- `Component` called `Seo` for meta data
- `Component` called `Image Section`
- `Component` called `Box` for layout spacing and formatting
-  - `Image Section` has arbitrary fields AND the `Box` component

I've connected them like so:
- - `Home Page` is linked to `Seo` and `Blocks` - `Blocks` is a `Dynamic Zone`
- - `Blocks` (amongst other) contain `Image Section` which in turn has `Box`

The `Dynamic Zone` on 
<img width="1271" alt="Screen Shot 2021-11-17 at 2 13 27 PM" src="https://user-images.githubusercontent.com/11737572/142198583-b9059079-c851-44cb-a6e0-31d268516559.png">

## The Issue

I am trying to query the components and it appears that the `Box` query is buggy. Perhaps I'm missing something, though, I'm not sure, because the query seems to fit the schema format. The same query works, when `box {}` is removed from the query.
<img width="913" alt="Screen Shot 2021-11-17 at 2 18 55 PM" src="https://user-images.githubusercontent.com/11737572/142199312-07ead162-fec7-4730-87fc-c4b577b231ed.png">

### Here is a working version (note - this is without the  `box {}` field)
<img width="1791" alt="Screen Shot 2021-11-17 at 2 22 51 PM" src="https://user-images.githubusercontent.com/11737572/142199845-a4ba03f4-cafe-47a7-a638-c75ee427f779.png">

### Code snippets
```
query {
  homePage {
    data {
      attributes {
        seo {
          id
          title
        }
        blocks {
          ... on ComponentBlocksImageSection {
            src
            body
            box {
              mx
            }
          }
        }
      }
    }
  }
}
```

If applicable, add code samples to help explain your problem.

### System

- Strapi version: `4-beta.13`
- Database: `Postgres`

**Comment by ScottAgirs** [comment]
PS. 
Just fyi - have tested with various other components (with same structure - nested components), that yields same result.

**Comment by Convly** [comment]
Hello, it should be fixed by #11629 in the next release. Please reopen the issue if it persists :slightly_smiling_face:
